lib/bigquery/dataset.js

```diff
@@ -46,20 +53,121 @@ var streamRouter = require('../common/stream-router.js');
  *
  * @alias module:bigquery/dataset
  * @constructor
+ *
+ * @example
+ * var gcloud = require('gcloud');
+ *
+ * var bigquery = gcloud.bigquery({
+ *   keyFilename: '/path/to/keyfile.json',
+ *   projectId: 'grape-spaceship-123'
+ * });
+ * var dataset = bigquery.dataset('institutions');
  */
 function Dataset(bigQuery, id) {
+  var methods = {
+    /**
+     * Create a dataset.
+     *
+     * @example
+     * dataset.create(function(err, dataset, apiResponse) {
```

Comment:
Do we want to document the params here? Or is this something that we won't really be advertising and it's just going to be masked behind `autoCreate`?